packages/services/schema/src/lib/compose.ts

```diff
@@ -0,0 +1,427 @@
+import { createHmac } from 'node:crypto';
```

Comment:
This file was all existing code in `orchestrator.ts`, but I found that file overwhelming and wanted to split it up a bit.
If we don't want these changes alongside the util library usage, then I can submit a separate PR or we can opt not to move this code..

Reply:
It makes it definitely harder to review the actual relevant changes. 😓